validation/provisioning/rke2/provisioning_node_driver_test.go

```diff
@@ -108,22 +114,33 @@ func (r *RKE2NodeDriverProvisioningTestSuite) TestProvisioningRKE2Cluster() {
 			continue
 		}
 
-		provisioningConfig := *r.provisioningConfig
-		provisioningConfig.MachinePools = tt.machinePools
+		for _, cattleConfig := range r.cattleConfigs {
+			clusterConfig := new(clusters.ClusterConfig)
+			operations.LoadObjectFromMap(permutationdata.ClusterConfigKey, cattleConfig, clusterConfig)
+			require.NotNil(r.T(), clusterConfig.Provider)
 
-		if !slices.Contains(provisioningConfig.Providers, "vsphere") && tt.isWindows {
-			r.T().Skip("Windows test requires access to vsphere")
-		}
+			clusterConfig.MachinePools = tt.machinePools
+
+			name := tt.name + " Node Provider: " + clusterConfig.NodeProvider + " Kubernetes version: " + clusterConfig.KubernetesVersion + " cni: " + clusterConfig.CNI
+			r.Run(name, func() {
+				if clusterConfig.Provider != "vsphere" && tt.isWindows {
+					r.T().Skip("Windows test requires access to vsphere")
+				}
 
-		permutations.RunTestPermutations(&r.Suite, tt.name, tt.client, &provisioningConfig, permutations.RKE2ProvisionCluster, nil, nil)
+				provider := provisioning.CreateProvider(clusterConfig.Provider)
+				credentialSpec := cloudcredentials.LoadCloudCredential(string(provider.Name))
+				machineConfigSpec := machinepools.LoadMachineConfigs(string(provider.Name))
+
+				clusterObject, err := provisioning.CreateProvisioningCluster(tt.client, provider, credentialSpec, clusterConfig, machineConfigSpec, nil)
+				require.NoError(r.T(), err)
+
+				provisioning.VerifyCluster(r.T(), tt.client, clusterConfig, clusterObject)
```

Comment:
https://github.com/rancher/tests/blob/9310be29e1e9cadcc54b0fc5e20824eacf2dcffe/actions/provisioning/permutations/permutations.go#L156 is missing in tests that use new permutations. If the cloud provider was enabled, it should be verified.

Reply:
one small nit here, discussed offline